src/main/java/com/teamcautionrobotics/robot/DriveBase.java

```diff
@@ -61,6 +75,46 @@ public void drive(double speed) {
         drive(speed, speed);
     }
 
+    public double getJerkLimit() {
+        return jerkLimit;
+    }
+
+    public void setJerkLimit(double jerkLimit) {
+        this.jerkLimit = jerkLimit;
+    }
+
+    public void driveSmoothly(double leftInput, double rightInput) {
+        updateDerivatives(leftInput, rightInput);
+        double leftPower = leftInput;
+        double rightPower = rightInput;
+
+        // limit jerk for each side if predicted jerk is too high
+        if (leftInputDerivative > jerkLimit) {
+            leftPower = limitJerk(leftInput, lastLeftPower);
+        }
+
+        if (rightInputDerivative > jerkLimit) {
+            rightPower = limitJerk(rightInput, lastRightPower);
+        }
+        drive(leftPower, rightPower);
+
+        lastLeftPower = leftPower;
+        lastRightPower = rightPower;
+    }
+
+    private void updateDerivatives(double leftInput, double rightInput) {
+        leftInputDerivative = (leftInput - lastLeftPower) / dt;
+        rightInputDerivative = (rightInput - lastRightPower) / dt;
+    }
+
+    // make the actual jerk = the threshold
+    private double limitJerk(double input, double lastPower) {
```

Comment:
This function is poorly named, since it does not as much limit jerk, as it just applies the maximum permissible jerk to the output value.